src/sentry/static/sentry/app/utils/apm.jsx

```diff
@@ -9,20 +9,20 @@ function startTransaction() {
   // times. This would result in losing the start of the transaction.
   Sentry.configureScope(scope => {
     if (firstPageLoad) {
-      firstPageLoad = false;
-    } else {
-      const prevTransactionSpan = scope.getSpan();
-      // If there is a transaction we set the name to the route
-      if (prevTransactionSpan && prevTransactionSpan.timestamp === undefined) {
-        prevTransactionSpan.finish();
-      }
-      scope.setSpan(
-        Sentry.startSpan({
-          op: 'navigation',
-          sampled: true,
-        })
-      );
+      return;
+    }
+
+    const prevTransactionSpan = scope.getSpan();
+    // If there is a transaction we set the name to the route
+    if (prevTransactionSpan && prevTransactionSpan.timestamp === undefined) {
```

Comment:
We already performs `if (typeof timestamp === 'number) return` check in the `finish` itself so it can be skipped here

Reply:
👍